pkg/util/informermanager/single-cluster-manager.go

```diff
@@ -98,5 +117,17 @@ func (s *singleClusterInformerManagerImpl) Start(stopCh <-chan struct{}) {
 }
 
 func (s *singleClusterInformerManagerImpl) WaitForCacheSync(stopCh <-chan struct{}) map[schema.GroupVersionResource]bool {
-	return s.informerFactory.WaitForCacheSync(stopCh)
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	res := s.informerFactory.WaitForCacheSync(stopCh)
+	for resource, synced := range res {
+		_, exist := s.syncedInformers[resource]
+		if !exist && synced {
+			s.syncedInformers[resource] = sets.Empty{}
+		} else if exist && !synced {
+			// may happen when resyncing
+			delete(s.syncedInformers, resource)
+		}
```

Comment:
Seems the sync state is unrelated to "resync". Do you provide more evidence?

Reply:
That's right, I have just taken a careful look of `client-go`. This condition may never happen.